src/main.rs

```diff
@@ -99,18 +102,20 @@ pub extern "C" fn load_elf(kernel_start: PhysAddr, kernel_size: u64,
 
 #[lang = "panic_fmt"]
 #[no_mangle]
-pub extern fn rust_begin_panic(_msg: core::fmt::Arguments,
+pub extern fn rust_begin_panic(msg: core::fmt::Arguments,
                                _file: &'static str,
                                _line: u32,
                                _column: u32) -> ! {
-    const VGA_BUFFER: *mut u8 = 0xb8000 as *mut _;
-
-    unsafe {
-        let vga_buffer = slice::from_raw_parts_mut(VGA_BUFFER, 25 * 80 *2);
-        vga_buffer[0] = b'E'; vga_buffer[1] = 0x4f;
-        vga_buffer[2] = b'R'; vga_buffer[3] = 0x4f;
-        vga_buffer[4] = b'R'; vga_buffer[5] = 0x4f;
-    }
+    use core::fmt::Write;
+    write!(printer::PRINTER.lock(), "ERR: {}", msg).unwrap();
+    // const VGA_BUFFER: *mut u8 = 0xb8000 as *mut _;
+
+    // unsafe {
+    //     let vga_buffer = slice::from_raw_parts_mut(VGA_BUFFER, 25 * 80 *2);
+    //     vga_buffer[0] = b'E'; vga_buffer[1] = 0x4f;
+    //     vga_buffer[2] = b'R'; vga_buffer[3] = 0x4f;
+    //     vga_buffer[4] = b'R'; vga_buffer[5] = 0x4f;
+    // }
```

Comment:
Could you remove this commented out part?

Reply:
Sure, one sec.